server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -205,7 +205,7 @@ public CachingClusteredClient(
 
       // Pull cached segments from cache and remove from set of segments to query
       final Map<Cache.NamedKey, byte[]> cachedValues;
-      if (useCache) {
+      if (useCache && cacheKeys.size() <= cacheConfig.getCacheBulkMergeLimit()) {
```

Comment:
Also, if the cacheBulkMergeLimit exceeds and populateCache is set to true, this will still ask the historicals for bySegment results and try to populate and merge them, we should also disable cache population in this case ?